src/backend/distributed/executor/multi_utility.c

```diff
@@ -2921,6 +2934,126 @@ ErrorIfUnsupportedTruncateStmt(TruncateStmt *truncateStatement)
 }
 
 
+/*
+ * ProcessTruncateStatement handles distributed locking
```

Comment:
Shall we mention that the actual `TRUNCATE` is done on the trigger and maybe the reasoning behind leaving that there?

Reply:
added